### DISCUSSION TITLE
trivy aws : support for --profile flag

### DISCUSSION THREAD

**Discussion by vishnukumarkvs** [aquasecurity/trivy#6419] [Ideas]
### Description

Requesting support for `--profile` flag for **trivy aws** CLI tool. 

The `--profile` support would enable users to specify named profiles containing the necessary credentials and settings for authenticating against AWS services.

### Proposed Changes

I am able to add aws profile support in my local codebase. This requires changes to two packages.
- aquasecurity/trivy-aws
- aquasecurity/trivy

As trivy uses **trivy-aws** for scanner, I first added aws profie support to trivy-aws package and then was able to add profile support to trivy

### Benefits
- **Improved Usability**: Users can switch between AWS accounts and roles effortlessly using named profiles.
- **Consistency with AWS Tools**: Aligns Trivy AWS CLI with existing AWS CLI workflows, reducing the learning curve for users.

### Contribution

I'm excited to contribute to this enhancement and have already made the necessary changes. Let me know if I can create a pull request.

### Screenshots

**Without profile. Uses default profile.**

![s1](https://github.com/aquasecurity/trivy/assets/116954249/a37661e1-04e1-450e-afe7-fc3f0bace88d)
![s2](https://github.com/aquasecurity/trivy/assets/116954249/62a8b5f6-87b2-45ff-8e04-142dc83a9a74)

**With --profile flag**

![s3](https://github.com/aquasecurity/trivy/assets/116954249/dd461c9d-56ea-48f8-8508-66d0ae8be84c)
![s4](https://github.com/aquasecurity/trivy/assets/116954249/7eeb28e4-cc72-4882-88e5-eb58a5081c19)

Looking forward to your input and collaboration!

### Target

AWS

### Scanner

None

**Comment by nikpivkin** [comment]
Hi @vishnukumarkvs !

You can set the `AWS_PROFILE` environment variable. https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-using-profiles

**Reply by vishnukumarkvs** [reply]
Hi @nikpivkin ,

Thanks. I've checked the documentation present at https://github.com/aquasecurity/trivy/blob/main/docs/docs/target/aws.md.  I couldn't find any information regarding switching profiles. As profiles are commonly used, would it be possible to add this to the documentation?

Using environment variables works but I believe that the `--profile` command line option is widely utilized. If you're open to it, I am happy to contribute by adding this flag.

Nonetheless, thanks for resolving my issue.

**Reply by nikpivkin** [reply]
@vishnukumarkvs We use the AWS SDK, so we support all environment variables that are specified in the [documentation](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/). I think we can clarify this in the documentation. 
I'm not sure about the need for this flag. @simar7 WDYT?

**Reply by simar7** [reply]
As @nikpivkin mentioned, we already support all environment variables that the AWS SDK offers, so I don't think there's a need for an additional flag at this time.
